### ISSUE TITLE
Add literals holding original text to ThematicBreak and HardLineBreak

### ISSUE THREAD

**Issue by Jeyhey** [commonmark/commonmark-java#295]
Currently it is not possible to know which text was parsed to ThematicBreak. Was it --- or was it *** ? This information is lost when parsing mark up text in this library.

In order to enable the user of the library to differentiate, it would be useful to keep the text literal in a variable of ThematicBreak. So one could do (in Kotlin)

```
// prints the original text that was parsed
fun print(node: Node): String {
    return when (node) {
        is ThematicBreak -> node.literal
        ...
    }
}
```

The same issue exists with HardLineBreak where two or more line breaks are reduced to one HardLineBreak-element. Adding a literal variable would enable to reconstruct the original text that was parsed and one could check how many line breaks the text originally had.

More generally, this feature request goes into the direction of enabling a bijection between the text and its parsed representation. I am not sure if a full bijection makes sense, but for the two node types of ThematicBreak and HardLineBreak it definitely does.

**Comment by robinst** [comment]
Thanks, done for `ThematicBreak`. Can you create a separate issue for `HardLineBreak` please? I'm not sure what you mean by "The same issue exists with HardLineBreak where two or more line breaks are reduced to one HardLineBreak-element".

On printing the original text that was parsed, have you seen this recently merged pull request?:
* https://github.com/commonmark/commonmark-java/pull/306

(To be released in the next release soon.)

**Comment by robinst** [comment]
Released in 0.22.0 now: https://github.com/commonmark/commonmark-java/blob/commonmark-parent-0.22.0/CHANGELOG.md#0220---2024-03-15
